gnucash/gnome-utils/dialog-properties.c

```diff
@@ -0,0 +1,1702 @@
+
+//    gnc_props_split_widget_name (gtk_buildable_get_name(GTK_BUILDABLE(fb)), &path);
+
+    g_free (path);
+
+    gtk_widget_show_all (GTK_WIDGET(fb));
+}
+
+/****************************************************************************/
+
+/** Connect a GtkRadioButton widget to its properties value.
+ *
+ *  @internal
+ *
+ *  @param button A pointer to the radio button that should be
+ *  connected.
+ */
+static void
+gnc_props_connect_radio_button (GtkRadioButton *button)
+{
+    gchar *path = NULL;
+
+    g_return_if_fail (GTK_IS_RADIO_BUTTON(button));
+
+//    gnc_props_split_widget_name (gtk_buildable_get_name(GTK_BUILDABLE(button)), &path);
+
+    g_free (path);
+}
+
+/****************************************************************************/
+
+/** Connect a GtkCheckButton widget to its properties value.
+ *
+ *  @internal
+ *
+ *  @param button A pointer to the check button that should be
+ *  connected.
+ */
+static void
+gnc_props_connect_check_button (GtkCheckButton *button)
+{
+    gchar *path;
+    const gchar *text;
+    gboolean active = FALSE;
+
+    g_return_if_fail (GTK_IS_CHECK_BUTTON(button));
+
+    gnc_props_split_widget_name (gtk_buildable_get_name(GTK_BUILDABLE(button)), &path);
+
+    text = qof_book_get_string_option (gnc_get_current_book(), path);
+
+    if (text)
+        g_object_set_data_full (G_OBJECT(button), "old-value", g_strdup (text), g_free);
+
+    if (g_strcmp0 (text, "t") == 0)
+        active = TRUE;
+
+    gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON(button), active);
+
+    g_free (path);
+}
+
+static void
+gnc_props_save_check_button (GtkCheckButton *button)
+{
+    gchar *path;
+    const gchar *old_value = g_object_get_data (G_OBJECT (button), "old-value");
+    gboolean active = gtk_toggle_button_get_active (GTK_TOGGLE_BUTTON(button));
+    gchar *new_value = active ? "t" : "";
+
+    gnc_props_split_widget_name (gtk_buildable_get_name(GTK_BUILDABLE(button)), &path);
+
+    if (g_strcmp0 (new_value, old_value) != 0)
+        qof_book_set_string_option (gnc_get_current_book(), path, new_value);
+
+    g_free (path);
+}
+
+/****************************************************************************/
+
+/** Connect a GtkSpinButton widget to its properties value.
+ *
+ *  @internal
+ *
+ *  @param button A pointer to the spin button that should be
+ *  connected.
+ */
+static void
+gnc_props_connect_spin_button (GtkSpinButton *spin)
+{
+    gchar *path;
+    gdouble vald = 0;
+    gint    val = 0;
+
+    g_return_if_fail (GTK_IS_SPIN_BUTTON(spin));
+
+    gnc_props_split_widget_name (gtk_buildable_get_name(GTK_BUILDABLE(spin)), &path);
+
+    val = qof_book_get_int64_option (gnc_get_current_book(), path);
+
+    // some spin values are doubles
+    if (val == 0)
+        vald = qof_book_get_double_option (gnc_get_current_book(), path);
+
+    if ((vald > 0) || g_strcmp0 (path, "options/Accounts/Day Threshold for Read-Only Transactions (red line)") == 0)
```

Comment:
Make a separate `gnc_props_connect_spin_button_double`. Hard-coding the path here is asking for maintenance trouble later.

Reply:
I realised that but could not think of an alternative till now by specifying double in the glade file widget name.